File: app/src/main/java/com/example/influencer/UI/OnBoarding/OnBoardingActivity.java
package com.example.influencer.UI.OnBoarding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.example.influencer.R;
import com.example.influencer.UI.Login.LoginActivity;
import com.example.influencer.UI.OnBoarding.Fragments.OnboardingFragment1;
import com.example.influencer.UI.OnBoarding.Fragments.OnboardingFragment2;
import com.example.influencer.UI.OnBoarding.Fragments.OnboardingFragment3;
import com.example.influencer.databinding.ActivityOnboardingBinding;
import com.tbuonomo.viewpagerdotsindicator.DotsIndicator;

public class OnBoardingActivity extends AppCompatActivity {

    private ActivityOnboardingBinding binding;
    private OnboardingViewModel viewModel;
    private DotsIndicator dotsIndicator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityOnboardingBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        viewModel = new ViewModelProvider(this).get(OnboardingViewModel.class);

        // Set up ViewPager2 adapter
        OnboardingPagerAdapter pagerAdapter = new OnboardingPagerAdapter(this);
        pagerAdapter.addFragment(new OnboardingFragment1());
        pagerAdapter.addFragment(new OnboardingFragment2());
        pagerAdapter.addFragment(new OnboardingFragment3());
        binding.onboardingViewPager.setAdapter(pagerAdapter);

        dotsIndicator = binding.dotsIndicator;
        dotsIndicator.attachTo(binding.onboardingViewPager);

        binding.buttonStartNow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewModel.StartnowSelected();
            }
        });

        viewModel.navigateToHome.observe(this, event -> {
            if (event != null && event.getContentIfNotHandled() != null) {
                goTonavigateToHome();
            }
        });
    }

    private void goTonavigateToHome() {
        //hace algo
    }
}

File: app/src/main/java/com/example/influencer/Features/OnBoarding___CountryAndDateSelector/UI/OnBoarding/OnBoardingActivity.kt
package com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModel
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.CountryAndDateSelector.CountryDateActivity
import com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding.Fragments.OnboardingFragment1
import com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding.Fragments.OnboardingFragment2
import com.example.influencer.Features.OnBoarding___CountryAndDateSelector.UI.OnBoarding.Fragments.OnboardingFragment3
import com.example.influencer.databinding.ActivityOnboardingBinding
import com.tbuonomo.viewpagerdotsindicator.DotsIndicator
import dagger.hilt.android.AndroidEntryPoint
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@AndroidEntryPoint
class OnBoardingActivity : AppCompatActivity() {

    private lateinit var binding: ActivityOnboardingBinding
    private val viewModel: OnboardingViewModel by viewModels()
    private lateinit var dotsIndicator: DotsIndicator

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityOnboardingBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupViewPager()
        setupDotsIndicator()
        setupStartNowButton()
        observeNavigationEvents()
    }

    private fun setupViewPager() {
        val pagerAdapter = OnboardingPagerAdapter(this).apply {
            addFragment(OnboardingFragment1())
            addFragment(OnboardingFragment2())
            addFragment(OnboardingFragment3())
        }
        binding.onboardingViewPager.adapter = pagerAdapter
    }

    private fun setupDotsIndicator() {
        dotsIndicator = binding.dotsIndicator
        dotsIndicator.attachTo(binding.onboardingViewPager)
    }

    private fun setupStartNowButton() {
        binding.buttonStartNow.setOnClickListener {
            viewModel.startNowSelected()
        }
    }

    private fun observeNavigationEvents() {
        viewModel.navigateToHome.observe(this) { event ->
            event.getContentIfNotHandled()?.let {
                navigateToHome()
            }
        }
    }

    private fun navigateToHome() {
        val intent = Intent(this, CountryDateActivity::class.java)
        startActivity(intent)
        finish()
    }
}